docs/src/modules/utils/CodeCopy.tsx

```diff
@@ -160,30 +160,34 @@ export function CodeCopyProvider({ children }: CodeCopyProviderProps) {
   const rootNode = React.useRef<HTMLDivElement | null>(null);
   React.useEffect(() => {
     document.addEventListener('keydown', (event) => {
-      if (hasNativeSelection(event.target as HTMLTextAreaElement)) {
-        // Skip if user is highlighting a text.
+      if (!rootNode.current) {
         return;
       }
-      // event.key === 'c' is not enough as alt+c can lead to ©, ç, or other characters on macOS.
-      // event.code === 'KeyC' is not enough as event.code assume a QWERTY keyboard layout which would
-      // be wrong with a Dvorak keyboard (as if pressing J).
-      const isModifierKeyPressed = event.ctrlKey || event.metaKey || event.altKey;
-      if (String.fromCharCode(event.keyCode) !== 'C' || !isModifierKeyPressed) {
+
+      // Skip if user is highlighting a text.
+      if (hasNativeSelection(event.target as HTMLTextAreaElement)) {
         return;
       }
-      if (!rootNode.current) {
+
+      // Skip if it's not a copy keyboard event
+      if (
+        !(
+          (event.ctrlKey || event.metaKey) &&
+          event.key.toLowerCase() === 'c' &&
+          !event.shiftKey &&
+          !event.altKey
+        )
+      ) {
         return;
       }
+
       const copyBtn = rootNode.current.querySelector('.MuiCode-copy') as HTMLButtonElement | null;
-      if (!copyBtn) {
-        return;
-      }
-      const initialEventAction = copyBtn.getAttribute('data-ga-event-action');
+      const initialEventAction = copyBtn!.getAttribute('data-ga-event-action');
```

Comment:
If we are doing `copyBtn!` on the line just below, the why do `as HTMLButtonElement | null`?
Wouldn't it be better to just do `as HTMLButtonElement` here and remove all the exclamation marks below?